packages/core/src/common/resource.ts

```diff
@@ -224,29 +224,54 @@ export class DefaultResourceProvider {
 
 }
 
+export type ResourceInitializationOptions = Pick<Resource, 'autosaveable' | 'initiallyDirty' | 'readOnly'>
```

Comment:
This is hard to understand as opposed to a simple interface type.

Reply:
It's also guaranteed to maintain the correct type if anything changes in `Resource`. For example, `readonly` had changed from `boolean` to `boolean | MarkdownString` since the original declaration was written, and the declaration hadn't kept up.